internal/pipeline/plugins.go

```diff
@@ -31,6 +31,10 @@ func (p *Plugins) UnmarshalOrdered(o any) error {
 	}
 
 	switch o := o.(type) {
+	case nil:
+		*p = Plugins{}
```

Comment:
i'm rethinking this a little - this makes nil unmarshal to an empty slices, but it should probably unmarshal to nil, right? all slices being reference types and all that? does it matter that much? they both work the same.

Reply:
I would write `nil` for these reasons:

- laziness
- when I want to remind the reader that `nil` is a valid slice value
- in order to distinguish "missing" instead of merely "empty"

I would write `[]something{}` for these reasons:

- when I want to remind the reader that checking for empty slices by comparing to `nil` instead of `len(slice) == 0` is not good enough
- in order to distinguish "not missing but has no items"
- maybe the slice has to pass through an interface, in which case untyped `nil` has a different meaning

Reply:
I think nil -> nil is consistent with the old behaviour before any pipeline parsing was done by the agent, yeah? So let's do that.

Reply:
> in order to distinguish "missing" instead of merely "empty"

i think this one is the key reason to use nil - in this case, plugins are missing and not empty, and the empty case should be handled by the `*ordered.MapSA` branch in the typeswitch.